File: libaurum/src/Accessibility/AccessibleWindow.cc
#include "AccessibleWindow.h"
#include "AccessibleWatcher.h"

AccessibleWindow::AccessibleWindow(std::shared_ptr<AccessibleApplication> app, std::shared_ptr<AccessibleNode> node)
: mApp{app}, mNode{node}
{
}

AccessibleWindow::~AccessibleWindow()
{
}

std::string AccessibleWindow::getTitle(void)
{
    return mNode->getText();
}

bool AccessibleWindow::isShowing()
{
    return mNode->isShowing();
}

bool AccessibleWindow::isActive()
{
    return mNode->isActive();
}

std::shared_ptr<AccessibleApplication> AccessibleWindow::getApplication(void)
{
    return mApp;
}

std::shared_ptr<AccessibleNode> AccessibleWindow::getNode(void)
{
    return mNode;
}

File: AurumSharp/AurumSharp/AccessibleWindow.cs
using System;

namespace SharpTest.libsharptest
{
    public class AccessibleWindow
    {
        private AccessibleApplication mApp = null;

        private AccessibleNode mNode = null;

        public AccessibleWindow(AccessibleApplication app, AccessibleNode node)
        {
            mApp = app;
            mNode = node;
        }

        internal String getTitle()
        {
            return mNode.getText();
        }

        internal bool isShowing()
        {
            return mNode.isShowing();
        }

        internal bool isActive()
        {
            return mNode.isActive();
        }

        internal AccessibleApplication getApplication()
        {
            return mApp;
        }

        internal AccessibleNode getAccessibleNode()
        {
            return mNode;
        }



    }
}